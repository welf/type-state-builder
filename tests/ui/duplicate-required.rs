use type_state_builder::TypeStateBuilder;

#[derive(TypeStateBuilder)]
struct DuplicateRequired {
    // This should be an error: duplicate required attributes
    #[builder(required)]
    #[builder(required)]
    name: String,

    email: String,
}

fn main() {
    let _instance = DuplicateRequired::builder()
        .name("test".to_string())
        .email("test@example.com".to_string())
        .build();
}