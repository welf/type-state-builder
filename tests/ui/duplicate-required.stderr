error: Duplicate required attribute. Only one required is allowed per field
 --> tests/ui/duplicate-required.rs:7:15
  |
7 |     #[builder(required)]
  |               ^^^^^^^^

error[E0599]: no function or associated item named `builder` found for struct `DuplicateRequired` in the current scope
  --> tests/ui/duplicate-required.rs:14:40
   |
4  | struct DuplicateRequired {
   | ------------------------ function or associated item `builder` not found for this struct
...
14 |     let _instance = DuplicateRequired::builder()
   |                                        ^^^^^^^ function or associated item not found in `DuplicateRequired`
